server.go

```diff
@@ -73,13 +73,14 @@ func (s *BaseServer) process(request *Request) (*Response, error) {
 	return NewResponse(request.ID, msg, ret...)
 }
 
-func (s *BaseServer) worker(ctx context.Context, ch <-chan *Request, cb Callback) {
+func (s *BaseServer) worker(wg *sync.WaitGroup, ctx context.Context, ch <-chan *Request, cb Callback) {
```

Comment:
convention always put the `ctx context.Context` as first argument
```suggestion
func (s *BaseServer) worker(ctx context.Context, wg *sync.WaitGroup,  ch <-chan *Request, cb Callback) {
```

Reply:
sure